mage_ai/data_preparation/models/block/r/__init__.py

```diff
@@ -145,7 +157,13 @@ def __render_r_script(
             code=code,
             global_vars=global_vars_str,
             input_paths=[
-                os.path.join(v.variable_path, DATAFRAME_CSV_FILE)
+                os.path.join(
```

Comment:
why do we change to this path join?

Reply:
The expression `os.path.join(v.variable_path, DATAFRAME_CSV_FILE)` was reading from a path `/home/src/mage_data/............../sample_data/data.csv`, which missed the required `output_0/` in the subpath where the file was stored by the previous data loader block, i.e.,  `/home/src/mage_data/............../sample_data/output_0/data.csv`. Thus, the file path has to be updated to read from the right place.

Reply:
Please see if it needs to be changed to the following code, which would be easier to understand:
```
                os.path.join(
                    v.variable_path,
                    CHILD_DATA_VARIABLE_UUID,
                    DATAFRAME_CSV_FILE
                )
```

Reply:
@wangxiaoyou1993 identified and fixed the root cause (i.e., a much cleaner solution), please review the updated PR, thanks.